### PR TITLE
[ENH] Deprecate Filter transformer

### PR DESCRIPTION
**Pull request by TonyBagnall**
part of #1086 
see #1108 
Filter was ported in from aeon-neuro for unknown reasons, and introduces a soft dependency on mne which we dont really need.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Fine with this. Deprecation message was left and needs updating.

**Review by hadifawaz1999 — APPROVED** [review]
lgtm !

**Review by MatthewMiddlehurst — APPROVED** [review]

File: aeon/transformations/filter.py [hunk 1]
```diff
@@ -4,10 +4,18 @@
...
 from aeon.transformations.base import BaseTransformer
 
 
+# TODO: remove in v0.8.0
+@deprecated(
+    version="0.6.0",
+    reason="EAgglo will be removed from annotation in v0.8.0, it has been replaced by "
+    "EAggloSegmenter in the segmentation module.",
+    category=FutureWarning,
+)
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/filter.py:18]: Message needs to be changed.

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Oops above was an accident.

File: aeon/transformations/filter.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/transformations/filter.py:18]: huh, thought Id done that

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**